libraries/helpers.rb

```diff
@@ -30,6 +30,39 @@ def keys_from_url(url)
       end
     end
 
+    # Determines if the user's shell is valid on the machine, otherwise
+    # returns the default of /bin/sh
+    #
+    # @return [String]
+    def valid_shell(shell)
```

Comment:
I think we should use a different name for this method, it might even be worth breaking things into two different helpers.

- `shell_exists?(candidate_shell)` which will check (and return a boolean) if the candidate shell actually exists on the system
- `default_shell` which returns the best candidate default shell based on platform